### DISCUSSION TITLE
Need help resolving xUnit1044 for an Action<T>

### DISCUSSION THREAD

**Discussion by grosch-intl** [xunit/xunit#2941] [Question]
I have the following TheoryData creator and I'm not seeing any way I can make this work without triggering xUnit1044.  Is there a workaround for this one?

```csharp
public static TheoryData<Action<ChecklistSoftwareFirmware>, string> StringChecks() {
        var results = new TheoryData<Action<ChecklistSoftwareFirmware>, string>();

        var values = new string?[] {
            null, string.Empty, " "
        };

        foreach (var value in values) {
            results.Add(x => x.CompilerName = value, "Missing compiler name.");
            ...
        }

        return results;
    }
```

**Comment by bradwilson** [comment]
You cannot.

- `Action<>` is not serializable
- There's no way to wrap inside of something that is
- There's no good generalized alternative I can suggest

You should disable `xUnit1044` around these types that are unsolvable.
